app/models/namespaces/project_namespace.rb

```diff
@@ -34,5 +34,19 @@ def of_ancestors_and_self
     def self.sti_name
       'Project'
     end
+
+    def self_and_parents
+      namespaces = [self]
+      namespaces += parent.self_and_ancestors unless parent.type == 'User'
+      namespaces
+    end
+
+    def update_metadata_summary_by_update_service(metadata_to_subtract, metadata_to_add)
```

Comment:
I think this method can just be called `update_metadata_summary`. Also I think we need to rename these variables as they seem to imply that metadata is being removed or added, instead of the counts

Reply:
So the reason its `...by_update_service` is because in the next PR, I have like 4 other `update_metadata_summary` functions (namespace_transfer, namespace_deletion, sample_deletion etc.). They're all different enough where I believe they require their own functions (definitely an area I would love review and a fresh set of eyes). So can we make note of this point but keep it until the next PR review?
Agreed on the argument var names, will update in next commit.

Reply:
Changed arguments to `deleted_metadata` and `added_metadata` and the functions that update the summary counts to `subtract_from_metadata_summary_count` and `add_to_metadata_summary_count` in 398dc863c9111dce05cac4a08fa3f29ea205ad0e